unitTests/test_ImageFactory.cpp

```diff
@@ -0,0 +1,103 @@
+#include <image.hpp> // Unit under test
+
+/// \todo we should not need to include all these headers to be able to use the Factory
+#include <jpgimage.hpp>
+#include <bmpimage.hpp>
+#include <cr2image.hpp>
+#include <crwimage.hpp>
+#include <gifimage.hpp>
+#include <jp2image.hpp>
+#include <mrwimage.hpp>
+#include <orfimage.hpp>
+#include <pgfimage.hpp>
+#include <pngimage.hpp>
+#include <psdimage.hpp>
+#include <rafimage.hpp>
+#include <rw2image.hpp>
+#include <tgaimage.hpp>
+#include <tiffimage.hpp>
+#include <webpimage.hpp>
+
+#include <error.hpp> // Need to include this header for the Exiv2::Error exception
+
+#include <gtest/gtest.h>
+
+using namespace Exiv2;
+
+TEST(TheImageFactory, createsInstancesForSupportedTypesInMemory)
+{
+    // Note that the constructor of these Image classes take an 'create' argument
+    EXPECT_NO_THROW(ImageFactory::create(ImageType::jp2));
+    EXPECT_NO_THROW(ImageFactory::create(ImageType::jpeg));
+    EXPECT_NO_THROW(ImageFactory::create(ImageType::exv));
+    EXPECT_NO_THROW(ImageFactory::create(ImageType::pgf));
+    EXPECT_NO_THROW(ImageFactory::create(ImageType::png));
+}
+
+TEST(TheImageFactory, cannotCreateInstancesForSomeTypesInMemory)
+{
+    // Note that the constructor of these Image classes does not take an 'create' argument
+
+    EXPECT_THROW(ImageFactory::create(ImageType::bmp), Error);
+    EXPECT_THROW(ImageFactory::create(ImageType::cr2), Error);
```

Comment:
Any particular reason why you are using `EXPECT_THROW` instead of `ASSERT_THROW`?

Reply:
Sorry about that. At work we use other testing framework and we took the decision of using the similar variant to EXPECT rather than the one similar to ASSERT. I will change it.